app/controllers/api/v1/materials_controller.rb

```diff
@@ -249,4 +302,40 @@ def assign_to_class
 
     render json: {:message => message}, :status => status
   end
+
+  private
+
+  #
+  # Return a single material item.
+  #
+  # id      The item id. (An id of ExternalActivity, Investigation, etc)
+  # type    A supported material type as a string key present in
+  #         the @@supported_material_types map. This will map to a
+  #         ruby class of the appropiate type, which will be returned
+  #         and can be used by materials libs to convert into json
+  #         and be consumed by API clients. 
+  #
+  def get_materials_item(id, type)
+
+    begin
+
+      rubyclass = @@supported_material_types[type]
+
+      if rubyclass 
+        item = rubyclass.find(id)
+
+        if item
+          return item
+        end
+ 
+      else
+        Rails.logger.error "Invalid material type #{type}"
+      end
+    rescue ActiveRecord::RecordNotFound => rnf
+      Rails.logger.error "RecordNotFound Invalid material id #{id}"
```

Comment:
I think it would be best to let this exception go up to the caller.  So it can be handled in the controller action method just like in the other cases.